ui/src/reportAnIssue.ts

```diff
@@ -33,22 +41,30 @@ Product: ${vscode.env.appName}
 Product Version: ${vscode.version}
 Language: ${vscode.env.language}`;
 
-    if (parsedError.stack) {
-        body = body.concat(`
+    // Try repeatedly with fewer lines of stack until we have a URL of a reasonable length
+    // tslint:disable-next-line: no-increment-decrement
+    for (let stackLines: number = numberOfLines(stack); stackLines >= 0; --stackLines) {
+        let bodyWithStack: string = body;
+        if (stack) {
+            bodyWithStack = bodyWithStack.concat(`
 
-<details>
+            <details>
 <summary>Call Stack</summary>
 
 \`\`\`
-${parsedError.stack}
+${limitLines(stack, stackLines)}
 \`\`\`
 
 </details>`);
-    }
 
-    const baseUrl: string = bugsUrl || `https://github.com/Microsoft/${extensionName}/issues`;
-    const url: string = `${baseUrl}/new?body=${encodeURIComponent(body)}`;
+            const baseUrl: string = bugsUrl || `https://github.com/Microsoft/${extensionName}/issues`;
+            const url: string = `${baseUrl}/new?body=${encodeURIComponent(bodyWithStack)}`;
 
-    // tslint:disable-next-line:no-floating-promises
-    await openUrl(url);
+            if (url.length < maxUrlLength || stackLines === 0) {
```

Comment:
Not gonna lie - makes me nervous having this kind of code in something so common as "Report an Issue" without unit tests. For example, the risks include opening a bunch of links if we forget a `return` or opening zero links if `url.length < maxUrlLength || stackLines === 0` has any problems.

Perhaps you move `await openUrl(url);` outside of the loop and create a function called `addStackToBody` with a few tests?

Reply:
Ah, I had intended to write unit tests, just forgot.